beacon_node/network/src/sync/range_sync/batch.rs

```diff
@@ -62,16 +62,16 @@ impl<T: EthSpec> PendingBatches<T> {
         let peer_request = batch.current_peer.clone();
         self.peer_requests
             .entry(peer_request)
-            .or_insert_with(|| HashSet::new())
+            .or_insert_with(HashSet::new)
             .insert(request_id);
         self.batches.insert(request_id, batch)
     }
 
-    pub fn remove(&mut self, request_id: &RequestId) -> Option<Batch<T>> {
-        if let Some(batch) = self.batches.remove(request_id) {
+    pub fn remove(&mut self, request_id: RequestId) -> Option<Batch<T>> {
```

Comment:
This is interesting. Do you remember the clippy warn for this one?
I would have assumed references were fine here.

Reply:
Yup! We define `type RequestId = usize`. 
This is what the [Rust Reference](https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout) says about usize:

> usize and isize have a size big enough to contain every address on the target platform. For example, on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.

So we're passing in by reference something that is AT MOST 8 bytes long (i.e the size of a reference...).
If I were to write ```rust
pub fn remove(&mut self, request_id: &usize) -> Option<Batch<T>>
``` it would definitely seem odd I think! :D

Reply:
Of course. Thanks!